### DISCUSSION TITLE
scaffold gaps "NNNNNNNN"

### DISCUSSION THREAD

**Discussion by INFINITY1993** [ablab/spades#942] [Q&A] [answered]
Hi, there, 
In the tutorial document, the author stated that "Contigs and scaffolds format Contigs/scaffolds names in SPAdes output FASTA files have the following format: >NODE_3_length_237403_cov_243.207 Here 3 is the number of the contig/scaffold, 237403 is the sequence length in nucleotides and 243.207 is the k-mer coverage for the last (largest) k value used. Note that the k-mer coverage is always lower than the read (per-base) coverage. In general, SPAdes uses two techniques for joining contigs into scaffolds. First one relies on read pairs and tries to estimate the size of the gap separating contigs. The second one relies on the assembly graph: e.g. if two contigs are separated by a complex tandem repeat, that cannot be resolved exactly, contigs are joined into scaffold with a fixed gap size of 100 bp. Contigs produced by SPAdes do not contain N symbols."
For the second senario, could we know how much the gaps are between two contigs? as the gap size is important to determine the connections of two genes separated by the gap. For example, co-occurrence of  antibiotic resistance genes and mobile gene elements is defined that these two genes should has a gap less than 10kb.

**Comment by asl** [comment, marked as answer]
Hello

No, it is not possible. Otherwise assembler would not produce a scaffold, but emit a complete sequence. Think about a tandem repeat with unknown multiplicity. While it might be not possible to determine its multiplicity (it might be longer than your input fragment length), still we can "scaffold over" if there are no other alternatives.

**Reply by INFINITY1993** [reply]
Thanks for your explicit explanation.
